ui/src/app/shared/services/extensions-service.ts

```diff
@@ -61,6 +62,20 @@ function registerStatusPanelExtension(component: StatusPanelExtensionComponent,
     extensions.eventTarget.emit('statusPanel', ext);
 }
 
+function registerTopBarActionMenuExt(
+    component: TopBarActionMenuExtComponent,
+    title: string,
+    id: string,
+    flyout?: ExtensionComponent,
```

Comment:
Should this `flyout` be mandatory?  The button click should always result in a flyout sliding panel?

Reply:
yes, good catch.  it should be required, since button ext works  with flyout